desc/efit.py

```diff
@@ -0,0 +1,373 @@
+        # For bdry_dist equal 1, all quantities *_truncated are the
+        # same as quantities without _truncated
+        chiN_truncated = np.linspace(0, bdry_dist, ns)
+        q_spline_truncated = CubicSpline(chiN_truncated, q_spline(chiN_truncated))
+
+        psi_spline_truncated = q_spline_truncated.antiderivative()
+        psi_truncated = psi_spline_truncated(chiN_truncated)
+        Psi_b_truncated = psi_truncated[-1]  # boundary flux
+        psiN_truncated = psi_truncated / Psi_b_truncated
+        rho_truncated = np.sqrt(psiN_truncated)
+
+        pressure = SplineProfile(pressure_psi(psi_truncated), rho_truncated)
+        iota = SplineProfile(iota_psi(psi_truncated), rho_truncated)
+
+        nbdry = g["nbbbs"]
+        rbdry = g["rbbbs"]
+        zbdry = g["zbbbs"]
+        theta = np.linspace(0, 2 * np.pi, nbdry)  # rbbbs includes endpoints
+
+        grid = LinearGrid(theta=theta)
+        basis = DoubleFourierSeries(M=M, N=N, sym=False)
+        transform = Transform(grid, basis, build_pinv=True)
+        Rb_lmn = transform.fit(rbdry)
+        Zb_lmn = transform.fit(zbdry)
+
+        surf = FourierRZToroidalSurface(
+            R_lmn=Rb_lmn,
+            Z_lmn=Zb_lmn,
+            modes_R=basis.modes[:, 1:],
+            modes_Z=basis.modes[:, 1:],
+        )
+
+        inputs = {}
+
+        inputs["pressure"] = pressure
+        inputs["iota"] = iota
+        inputs["sym"] = False
+        inputs["surface"] = surf
+        inputs["NFP"] = int(1)
+        inputs["Psi"] = Psi_b
+        inputs["L"] = L
+        inputs["M"] = M
+        inputs["N"] = N
+
+        eq = Equilibrium(**inputs)
+
+        return eq
+
+    @classmethod
+    def efit_to_desc_input(  # noqa: C901 - fxn too complex
+        cls,
+        eq,
+        outfile,
+        objective="force",
+        optimizer="lsq-exact",
+        ftol=1e-2,
+        xtol=1e-2,
+        gtol=1e-6,
+        maxiter=100,
+        threshold=1e-8,
+    ):
+        """Generate a DESC input file from a DESC equilibrium object.
+
+        DESC will automatically choose continuation parameters
+
+        Parameters
+        ----------
+        eq: Equilibrium
+            DESC equilibrium object
+        outfile : str or path-like
+            name of the DESC input file to create
+        objective : str
+            objective type used in the input file
+        optimizer : str
+            type of optimizer
+        ftol : float
+            relative tolerance of the objective function f
+        xtol : float
+            relative tolerance of the state vector x
+        gtol : float
+            absolute tolerance of the projected gradient g
+        maxiter : int
+            maximum number of optimizer iterations per continuation step
+        threshold : float
+            Fourier coefficients below this value will be set to 0.
+        """
+        with open(outfile, "w+") as f:
+            header = "# DESC input file generated from an geqdsk file\n# "
```

Comment:
would be better to allow header to be passed into this file, have the default be "generated from a DESC equilibrium on XXX date with DESC version COMMIT_TAG"

as the fxn does not need efit it should not be mentioned